### ISSUE TITLE
Joplin Server:  first user password is invalid

### ISSUE THREAD

**Issue by xXZaretXx** [laurent22/joplin#4922]
Hi,
i noticed, when i create a new user  the password is always invalid until i "update" the account with a new password
So i need to double set the password.

## Environment

Joplin version: Docker:latest - 1.7.2
Platform:  Ubuntu Server
OS specifics: 

## Steps to reproduce

1.  login as admin
2.  create new user and set directly a password
3.  logout and try to login with the new user( invalid password or user) 
(4. after setting a new password - this works)

## Describe what you expected to happen
The password is valid...

**Comment by laurent22** [comment]
I thought that was fixed on latest, but maybe the fix hasn't been released yet.

**Comment by laurent22** [comment]
Hmm, yes it should have been released, so maybe there's another bug.

**Comment by nishantwrp** [comment]
Can I try to work on this issue?

**Comment by laurent22** [comment]
Yes, sure.

**Comment by nishantwrp** [comment]
Hi, I tried to look into this issue, and I was not able to reproduce this issue. I was able to successfully sign in after creating a new user.

**Comment by xXZaretXx** [comment]
Mhm strange - i tried it again, with and without special charakters, but i only get  
```
Error: Invalid username or password
    at SessionModel.<anonymous> (/home/joplin/packages/server/src/models/SessionModel.ts:28:20)
    at Generator.next (<anonymous>)
    at fulfilled (/home/joplin/packages/server/dist/models/SessionModel.js:5:58)
```
Even simple combinations like test@localhost:test

Only updating the password solves the problem

**Comment by nishantwrp** [comment]
Which code exactly are you running? 

I used the latest code on the dev branch.

**Comment by xXZaretXx** [comment]
> Which code exactly are you running?
> 
> I used the latest code on the dev branch. If you're running a docker release, maybe the docker release isn't up to date.

Yeah .. i mention it in my issue - i used the latest docker release - 1.7.2 (joplin/server)  - the last update is 3 month ago

**Comment by laurent22** [comment]
This is strange, it sounds like it's still the old code running.

**Comment by xXZaretXx** [comment]
> This is strange, it sounds like it's still the old code running.

I'm sorry but what is strange?
https://hub.docker.com/r/joplin/server/tags?page=1&ordering=last_updated
If you look up - the latest build is the 1.7.2.
If i read the server commits correctly ( nice comments btw) the double hash fix is done after the release from 1.7.2.
So it's  applied in 1.8 which doesn't exist prebuild in docker

**Comment by laurent22** [comment]
No you were right, there was two issues with password hashing, and the one you're mentioning indeed was only fixed after 1.7.2. So it means it should be fixed in v2.0 then.
